packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2958,7 +2958,7 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
     if (!isRepaintBoundary && _wasRepaintBoundary) {
       _needsPaint = false;
       _needsCompositedLayerUpdate = false;
-      owner?._nodesNeedingPaint.remove(this);
+      owner?._nodesNeedingPaint.removeWhere((RenderObject t) => t == this);
```

Comment:
My apology for yet another comment, but I just realized `identical` should be more efficient for not going through `operator ==`.
```suggestion
      owner?._nodesNeedingPaint.removeWhere((RenderObject t) => identical(t, this));
```